toffy/qc_comp_test.py

```diff
@@ -178,129 +181,170 @@ def test_compute_99_9_intensity():
 
-        # assert the fovs are correct
-        misc_utils.verify_same_elements(
-            provided_fovs=fovs,
-            nzm_fovs=nonzero_mean_batch['fov'].values
-        )
-        misc_utils.verify_same_elements(
-            provided_fovs=fovs,
-            ti_fovs=total_intensity_batch['fov'].values
-        )
-        misc_utils.verify_same_elements(
-            provided_fovs=fovs,
-            i99_9_fovs=intensity_99_9_batch['fov'].values
+        # define a sample qc_path to write to
+        qc_path = os.path.join(temp_dir, 'sample_qc_dir')
+
+        # bin file error check
+        with pytest.raises(FileNotFoundError):
+            qc_comp.compute_qc_metrics(
+                'bad_bin_path', fovs[0], panel_path, gaussian_blur
+            )
+
+        # panel file error check
+        with pytest.raises(FileNotFoundError):
+            qc_comp.compute_qc_metrics(
+                bin_file_path, fovs[0], 'bad_panel_path', gaussian_blur
+            )
+
+        # fov error check
+        with pytest.raises(FileNotFoundError):
+            qc_comp.compute_qc_metrics(
+                bin_file_path, 'bad_fov', panel_path, gaussian_blur
+            )
+
+        # first time: create new files, also asserts qc_path is created
+        qc_comp.compute_qc_metrics(
+            bin_file_path, fovs[0], panel_path, gaussian_blur
         )
 
-        # assert the chans are correct
-        misc_utils.verify_same_elements(
-            provided_chans=chans,
-            nzm_chans=nonzero_mean_batch.drop(columns='fov').columns.values
+        nonzero_mean_path = os.path.join(
+            bin_file_path, '%s_nonzero_mean_stats.csv' % fovs[0]
         )
-        misc_utils.verify_same_elements(
-            provided_chans=chans,
-            nzm_chans=total_intensity_batch.drop(columns='fov').columns.values
+        total_intensity_path = os.path.join(
+            bin_file_path, '%s_total_intensity_stats.csv' % fovs[0]
         )
-        misc_utils.verify_same_elements(
-            provided_chans=chans,
-            nzm_chans=intensity_99_9_batch.drop(columns='fov').columns.values
+        percentile_99_9_path = os.path.join(
+            bin_file_path, '%s_percentile_99_9_stats.csv' % fovs[0]
         )
 
+        # assert the QC files for fov-1-scan-1 were created
+        assert os.path.exists(nonzero_mean_path)
+        assert os.path.exists(total_intensity_path)
+        assert os.path.exists(percentile_99_9_path)
+
+        # read in the QC data
+        nm = pd.read_csv(nonzero_mean_path)
+        ti = pd.read_csv(total_intensity_path)
+        p99_9 = pd.read_csv(percentile_99_9_path)
+
+        # assert the column names are correct
+        assert list(nm.columns.values) == ['fov', 'channel', 'Non-zero mean intensity']
+        assert list(ti.columns.values) == ['fov', 'channel', 'Total intensity']
+        assert list(p99_9.columns.values) == ['fov', 'channel', '99.9 intensity value']
```

Comment:
Can this stuff be condensed into a for loop over the tuple of file names and column names?